src/GraphQL/Utilities/AstPrinter.cs

```diff
@@ -328,25 +336,54 @@ public AstPrintVisitor()
                 c.Field(x => x.Value);
                 c.Print(f =>
                 {
-                    var val = f.Arg(x => x.Value);
-                    if (!string.IsNullOrWhiteSpace(val?.ToString()) && !val.ToString().StartsWith("\"", StringComparison.InvariantCulture))
+                    var val = (string)f.Arg(x => x.Value);
+                    if (val == null)
+                        throw new InvalidOperationException("StringValue cannot contain null");
+                    var sb = new System.Text.StringBuilder(val.Length + 2);
+                    sb.Append('\"');
+                    foreach (var ch in val)
                     {
-                        val = $"\"{val}\"";
+                        if (ch < ' ')
+                        {
+                            if (ch == '\n')
+                                sb.Append("\\\n");
```

Comment:
```suggestion
                                sb.Append("\\n");
```

```
StringBuilder b = new StringBuilder();
            b.Append("abc");
            b.Append("\\\n");
            b.Append("def");
            var s = b.ToString();
```
yields 
```
abc\
def
```

but we need `abc\ndef`

Reply:
Please add tests for \n \r \t.

Reply:
Also what about other escaped symbols?
![изображение](https://user-images.githubusercontent.com/21261007/111386094-cdd20980-86bc-11eb-9442-595f09e674bd.png)

By the way I do not remember that there was `/` when I last read the specification.

Reply:
Good catch.  I should have added tests :-(

Reply:
I would not encode the forward slash.  There is no reason to; it makes reading the resulting string harder.  As for the backspace character and form-feed characters, I have not seen those characters used since I wrote code for DOS!  Probably there's some use with telnet now, but I think we can live with those character's unicode encodings here!

Reply:
> It is not required

See cut of the spec grammar above. It is required by the spec itself. Is not it so?

> we can live with those character's unicode encodings here

Why not do it as required in the specification?

Reply:
From parser:
![изображение](https://user-images.githubusercontent.com/21261007/111454585-da3e7c80-8725-11eb-9949-328dafb2afc3.png)
and
![изображение](https://user-images.githubusercontent.com/21261007/111454628-e75b6b80-8725-11eb-8a8d-3290221d505d.png)

![изображение](https://user-images.githubusercontent.com/21261007/111454671-f3472d80-8725-11eb-81ae-8b807f5dd82a.png)

So forward slash **should** be escaped to work correctly. I'll add tests.

Reply:
Well, I figured out that our parser works will with both options, both `"ab/cd"` and `"ab\/cd"` are parsed as `"ab/cd"`. But I think that we should print 100% compatible AST that can be parsed by any parser so I added forward slash to escaped symbols. Tests added.

Reply:
Reread the spec. String character is any source character the except three noted. That includes a forward slash.

Reply:
So printing a forward slash is 100% valid in any parser guaranteed.

Reply:
The parser MUST parse all characters because it may be present in a document. This is a serializer not a parser.

Reply:
I don't understand. Where is it?

Reply:
Look up “Source Text” in the spec. SourceCharacter is
`/[\u0009\u000A\u000D\u0020-\uFFFF]/`

Which includes a forward slash. So string character also includes a forward slash because it is not a backslash or quote or line terminator

Reply:
Aaa, I think I understood. SourceCharacter is `/[\u0009\u000A\u000D\u0020-\uFFFF]/`

Reply:
A forward slash MAY be escaped of course. 

And the backspace must be encoded as either `\b` or `\u0008` but not it is not required that it be one vs the other

Reply:
We can leave the changes you pushed by the way. I just personally wouldn’t encode the forward slash.

Reply:
So only `/` MAY BE unescaped. Right? And all other `" \ b f n r t` SHOULD BE escaped.

Reply:
Well, let's remove escaping for forward slash.

Reply:
> Well, let's remove escaping for forward slash.

Done

Reply:
Make some notes if you wish and release. 👍

Reply:
I didn't add notes; it's just really a bugfix and listed along with the many other bugfixes for 4.0 in the search-prs-by-milestone list.  Too busy today to spend much time on it unfortunately.

Reply:
ok